DMInRoblox/src/server/VM/init.lua

```diff
@@ -0,0 +1,252 @@
+				return vm.locals[ref.Id]
+			elseif (ref.Type == "Local") then
+				return vm.locals[ref.Id + vm.argCount]
+			elseif (ref.Type == "Field") then
+				local owner = vm:Pop()
+				
+				return owner:GetVar(ref.Id)
+			else
+				error("Unknown reference type \"" ..ref.Type.. "\"")
+			end
+		end,
+		AssignReference = function(vm: VM, ref: DMReference, value: any)
+			if (ref.Type == "Argument") then
+				vm.locals[ref.Id] = value
+			elseif (ref.Type == "Local") then
+				vm.locals[ref.Id + vm.argCount] = value
+			elseif (ref.Type == "Field") then
+				local owner = vm:Pop()
+				
+				owner:SetVar(ref.Id, value)
+			else
+				error("Unknown reference type \"" ..ref.Type.. "\"")
+			end
+		end,
+	}
+	
+	for argIndex = 1, #args do
+		vm.locals[argIndex - 1] = args[argIndex]
+	end
+	
+	while (vm.pc <= #vm.bytecode) do
+		local opcode = vm.bytecode[vm.pc]
+		
+		vm.pc += 1
+		if (opcode == 0x2) then --PushType
+			local objectTypeId = vm:ReadInt()
+			
+			vm:Push(objectTree.GetObjectType(objectTypeId))
+		elseif (opcode == 0x6) then --PushReferenceValue
+			local reference = vm:ReadReference()
+			
+			vm:Push(vm:GetReferenceValue(reference))
+		elseif (opcode == 0x8) then --Add
+			local b = vm:Pop()
+			local a = vm:Pop()
+			
+			local aType = typeof(a)
+			local bType = typeof(b)
+			if (aType == "number" and bType == "number") then
+				vm:Push(a + b)
+			else
+				error("Cannot add " ..aType.. " and " ..bType)
+			end
+		elseif (opcode == 0x9) then --Assign
+			local reference = vm:ReadReference()
+			local value = vm:Pop()
+			
+			vm:AssignReference(reference, value)
+		elseif (opcode == 0xA) then --Call
+			local procRef = vm:ReadReference()
+			local args = vm:Pop()
+			
+			local proc
+			if (procRef.Type == "GlobalProc") then
+				proc = objectTree.GetProc(procRef.Id)
+			else
+				error("Unknown proc reference type " .. procRef.Type)
+			end
+			
+			if (typeof(proc) == "function") then --Native proc
+				vm:Push(proc(args))
+			elseif (typeof(proc) == "table") then --DM proc
+				vm:Push(module.RunBytecode(proc.Bytecode, args))
+			end
+		elseif (opcode == 0xC) then --JumpIfFalse
+			local pos = vm:ReadInt()
+			local value = vm:Pop()
+			
+			if (not isTruthy(value)) then
+				vm.pc = pos + 1
+			end
+		elseif (opcode == 0xE) then --Jump
+			vm.pc = vm:ReadInt() + 1
+		elseif (opcode == 0x10) then --Return
+			returnValue = vm:Pop()
+			break
+		elseif (opcode == 0x28) then --Multiply
+			local a = vm:Pop()
+			local b = vm:Pop()
+			
+			local aType = typeof(a)
+			local bType = typeof(b)
+			if (aType == "number" and bType == "number") then
+				vm:Push(a * b)
+			else
+				error("Cannot multiply " ..aType.. " and " ..bType)
+			end
+		elseif (opcode == 0x2E) then --CreateObject
+			local args = vm:Pop()
+			local objectType = vm:Pop()
+			
+			
+			vm:Push(objectType:Create(args))
+		elseif (opcode == 0x37) then --PushArguments
+			local argCount = vm:ReadInt()
+			local namedCount = vm:ReadInt()
```

Comment:
```
warning[unused_variable]: namedCount is assigned a value, but never used
    ┌─ src/server/VM/init.lua:225:10
    │
225 │             local namedCount = vm:ReadInt()
    │                   ^^^^^^^^^^
```

Reply:
`vm:ReadInt()` needs to be called, but named arguments aren't supported here. I suppose I could cause an error if this isn't 0.

Reply:
`_namedCount` also is prudent

Reply:
but `assert(namedCount == 0)` does sound fine